doc/dev.md

```diff
@@ -0,0 +1,50 @@
+# Local development
+
+This document sets you up to use locally running instances of `ocaml-ci-server` and `ocaml-ci-web` to build an OCaml project that is in a repository owned by your GitHub user.
+
+### Setting up a GitHub App
+
+Since `ocaml-ci` is a GitHub App, create a GitHub App under your own user and point it to localhost via a webhook payload delivery service like [smee.io](https://smee.io).
+
+To do this, follow the instructions in [Setting up your development environment to create a GitHub App](https://docs.github.com/en/developers/apps/getting-started-with-apps/setting-up-your-development-environment-to-create-a-github-app) but when it comes to setting permissions for your app, set the following:
+
+```
+Metadata: Read
+Checks: Read and write
+Commit statuses: Read and write
+```
+
+### Running the server locally
+
+You will need the following:
+
+1. The GitHub App ID of the app you created
+2. The `pem` file containing the private key associated to the app
+3. A capability file that has been generated for you by an ocaml-ci admin. This gives you the ability to submit jobs to the ci cluster
```

Comment:
This needs to be more specific.

You need a capability file for submitting jobs to a cluster, in this case the main `ocaml-ci` cluster as documented in https://github.com/ocurrent/ocluster#admin